### ISSUE TITLE
Failed to compute transactions root, micro blocks missing

### ISSUE THREAD

**Issue by jgraef** [nimiq/core-rs-albatross#26]
This panic was captured on our DevNet (branch `devnet-rc3`):

```
logs/devnet_validator_9.log- WARN  staking_contract     | Slashing NQ98 LFT2 96K1 314V H2DT D3UE 9BME BY9R FNM4 with 3.39218
logs/devnet_validator_9.log- WARN  staking_contract     | Slashing NQ89 1551 J9H2 K8C0 4XMV SG37 PP38 V6RT E5N4 with 3.39218
logs/devnet_validator_9.log- WARN  staking_contract     | Slashing NQ76 0JQR LU02 9RHX ES9A TSSX CNGE UPYE KTFC with 3.39218
logs/devnet_validator_9.log- INFO  consensus            | Now at block #7168
logs/devnet_validator_9.log- WARN  staking_contract     | Slashing NQ08 EGR6 4UUA LRBE GV9Q 50Y1 2FDU CPA0 GH7C with 3.37895
logs/devnet_validator_9.log: ERROR panic                | thread 'tokio-runtime-worker-5' panicked at 'Failed to compute transactions root, micro blocks missing': src/libcore/option.rs:1185
logs/devnet_validator_9.log-stack backtrace:
logs/devnet_validator_9.log-   0: log_panics::init::{{closure}}
logs/devnet_validator_9.log-   1: std::panicking::rust_panic_with_hook
logs/devnet_validator_9.log-             at src/libstd/panicking.rs:468
logs/devnet_validator_9.log-   2: std::panicking::continue_panic_fmt
logs/devnet_validator_9.log-             at src/libstd/panicking.rs:373
logs/devnet_validator_9.log-   3: rust_begin_unwind
logs/devnet_validator_9.log-             at src/libstd/panicking.rs:302
logs/devnet_validator_9.log-   4: core::panicking::panic_fmt
logs/devnet_validator_9.log-             at src/libcore/panicking.rs:139
logs/devnet_validator_9.log-   5: core::option::expect_failed
logs/devnet_validator_9.log-             at src/libcore/option.rs:1185
logs/devnet_validator_9.log-   6: nimiq_block_production_albatross::BlockProducer::next_macro_header
logs/devnet_validator_9.log-   7: nimiq_block_production_albatross::BlockProducer::next_macro_block_proposal
logs/devnet_validator_9.log-   8: nimiq_validator::validator::Validator::produce_macro_block
logs/devnet_validator_9.log-   9: <futures::future::lazy::Lazy<F,R> as futures::future::Future>::poll
logs/devnet_validator_9.log-  10: futures::task_impl::Spawn<T>::poll_future_notify
logs/devnet_validator_9.log-  11: std::panicking::try::do_call
logs/devnet_validator_9.log-  12: __rust_maybe_catch_panic
logs/devnet_validator_9.log-             at src/libpanic_unwind/lib.rs:83
```

**Comment by jgraef** [comment]
I merged the macro block sync that changes the relevant function. Need to recheck if this still happens.

**Comment by jgraef** [comment]
Seems to be fixed with #16 
It doesn't make sense to keep this open, since the whole function, that likely caused the panic, has been rewritten in that PR.
